content/s1/ClassesInheritanceInterfaces/index.md

```diff
@@ -0,0 +1,235 @@
+/// </summary>
+public abstract class Shape {}
+public virtual double CalculateArea();
+```
+#### Circle.cs
+
+```csharp
+/// <summary>
+/// Écrire une classe Circle qui hérite de Shape et qui contient une propriété publique de type double appelée "Radius".
+/// </summary>
+public class Circle {}
+
+/// <summary>
+/// Écrire un constructeur qui prend un double radius et qui initialise Radius à radius.
+/// </summary>
+public Circle(double radius);
+
+/// <summary>
+/// Écrire une méthode CalculateArea() qui retourne l'aire du cercle.
+/// L'aire d'un cercle est calculée en faisant radius * radius * Math.PI
+/// </summary>
+public override double CalculateArea();
+
+/// <summary>
+/// Écrire une méthode ToString() qui retourne une string de la forme "Circle with radius {Radius}"
+/// </summary>
+public override string ToString();
+
+/// <summary>
+///  Écrire une méthode statique CreateRandom qui prend un int maxRadius et qui retourne un nouveau cercle avec un rayon aléatoire entre 0 et maxRadius.
+/// Si maxRadius est négatif, une exception `ArgumentOutOfRangeException` est levée.
+/// Sa surcharge sans paramètre doit créer un cercle avec un rayon aléatoire entre 1 et 100.
+/// </summary>
+/// <remarks>
+/// Pour générer un nombre aléatoire, vous pouvez utiliser Random.Next.
+/// </remarks>
+public static Circle CreateRandom(int maxRadius);
+public static Circle CreateRandom();
+```
+
+#### Rectangle.cs
+
+```csharp
+/// <summary>
+/// Écrire une classe Rectangle qui hérite de Shape et qui contient deux propriétés publiques : double Width et double Height.
+/// </summary>
+public class Rectangle {}
+
+/// <summary>
+/// Écrire un constructeur qui prend un double width et un double height et qui initialise Width à width et Height à height.
+/// </summary>
+public Rectangle(double width, double height);
+
+/// <summary>
+/// Écrire une méthode override CalculateArea() qui retourne l'aire du rectangle.
+/// L'aire d'un rectangle est calculée en faisant width * height
+/// </summary>
+public override double CalculateArea();
+
+/// <summary>
+/// Écrire une méthode override ToString() qui retourne une string de la forme "Rectangle with width {Width} and height {Height}"
+/// </summary>
+public override string ToString();
+
+/// <summary>
+///  Écrire une méthode statique CreateRandom qui prend deux doubles (maxWidth et maxHeight) et qui retourne un nouveau rectangle avec une largeur entre 1 et maxWidth et une hauteur aléatoire entre 1 et maxHeight.
+/// Si maxWidth ou maxHeight est négatif, une exception `ArgumentOutOfRangeException` est levée.
+/// Sa surcharge sans paramètre doit créer un rectangle avec une largeur et une hauteur aléatoire entre 1 et 100.
+/// </summary>
+/// <remarks>
+/// Pour générer un nombre aléatoire, vous pouvez utiliser Random.Next.
+/// </remarks>
+public static Rectangle CreateRandom(double maxWidth, double maxHeight);
+public static Rectangle CreateRandom();
+```
```

Comment:
```suggestion
```csharp
/// <summary>
/// Écrire une classe Rectangle qui hérite de Shape et qui contient deux propriétés publiques : double Width et double Height.
/// </summary>
public class Rectangle {}
```
```csharp
/// <summary>
/// Écrire un constructeur qui prend un double width et un double height et qui initialise Width à width et Height à height.
/// </summary>
public Rectangle(double width, double height);
```
```csharp
/// <summary>
/// Écrire une méthode override CalculateArea() qui retourne l'aire du rectangle.
/// L'aire d'un rectangle est calculée en faisant width * height
/// </summary>
public override double CalculateArea();
```
```csharp
/// <summary>
/// Écrire une méthode override ToString() qui retourne une string de la forme "Rectangle with width {Width} and height {Height}"
/// </summary>
public override string ToString();
```
```csharp
/// <summary>
///  Écrire une méthode statique CreateRandom qui prend deux doubles (maxWidth et maxHeight) et qui retourne un nouveau rectangle avec une largeur entre 1 et maxWidth et une hauteur aléatoire entre 1 et maxHeight.
/// Si maxWidth ou maxHeight est négatif, une exception `ArgumentOutOfRangeException` est levée.
/// Sa surcharge sans paramètre doit créer un rectangle avec une largeur et une hauteur aléatoire entre 1 et 100.
/// </summary>
/// <remarks>
/// Pour générer un nombre aléatoire, vous pouvez utiliser Random.Next.
/// </remarks>
public static Rectangle CreateRandom(double maxWidth, double maxHeight);
public static Rectangle CreateRandom();
```
```